### PR TITLE
Handle either tolerated_failure_count or tolerated_failure_percentage

### PR DESCRIPTION
**Pull request by agrare**
Follow-up to https://github.com/ManageIQ/floe/pull/282#discussion_r1792238478 to handle both ToleratedFailureCount and ToleratedFailurePercent present

### CODE REVIEW

File: spec/workflow/states/map_spec.rb [hunk 1]
```diff
@@ -263,6 +263,15 @@
           end
         end
       end
+
+      context "with both ToleratedFailureCount and ToleratedFailurePercentage" do
+        let(:tolerated_failure_percentage) { 50 }
+        let(:tolerated_failure_count)      { 1 }
```
Comment (Fryguy) [inline spec/workflow/states/map_spec.rb:269]: Can you add another spec showing when count might be tolerated but not percentage (i.e. the flipped case to this?)

**Comment by Fryguy** [comment]
My brain is having trouble with ||= with booleans for some reason, but I think this is good.

File: spec/workflow/states/map_spec.rb (hunk 1 above)
Comment (agrare) [reply spec/workflow/states/map_spec.rb]: :+1: done, also added a case where they are both present but neither high enough for state success

**Comment by agrare** [comment]
> My brain is having trouble with ||= with booleans for some reason, but I think this is good.

Yeah it is funny how used to `||=` being for `nil` we get :smile:

**Review by Fryguy — APPROVED** [review]

File: lib/floe/workflow/states/map.rb [hunk 2]
```diff
@@ -110,13 +110,18 @@ def success?(context)
           total      = contexts.count
 
           return true if num_failed.zero? || total.zero?
-          return true if tolerated_failure_percentage && tolerated_failure_percentage == 100
+          return false if tolerated_failure_count.nil? && tolerated_failure_percentage.nil?
```
Comment (kbrock) [inline lib/floe/workflow/states/map.rb:113]: can this be a configuration error and not a runtime error?

**Pull request merged**